C_image_processing/filters/negative_filter.c

```diff
@@ -45,13 +43,21 @@ int negative_filter(const char *inputFile, const char *outputFile)
     // Apply negative filter to each pixel
     unsigned char *p = imageData;
     unsigned char *q = newImageData;
-    for (int i = 0; i < height * width; i++)
+    for (int y = 0; y < height; y++)
     {
-        *q++ = 255 - *p++;
-    }
+        for (int x = 0; x < width; x++)
+        {
+            int index = (y * width + x) * (bitDepth / 8);
 
+            for (int channel = 0; channel < bitDepth / 8; channel++)
+            {
+                newImageData[index + channel] = 255 - imageData[index + channel];
+            }
+        }
+    }
     // Write image data to output file
     fwrite(headerInfo, sizeof(unsigned char), 54, fileOut);
```

Comment:
as previously "dark_filter.c" was changed from " HeaderInfo " to " Byte " would for it make sense to be changed here as well to be consistent ?

Or is there something I'm miss reading or understanding from the code represented.

Reply:
changed it in the `dark_filter.c` file so not needed.

Reply:
Understood!, well done